Sources/SwiftSVG/PresentationAttributes.swift

```diff
@@ -80,10 +80,9 @@ public extension PresentationAttributes {
                 return nil
             }
             
-            var fill = Fill()
-            fill.color = fillColor ?? "black"
-            fill.opacity = fillOpacity ?? 1.0
-            return fill
+            return Fill(color: fillColor,
+                        opacity: fillOpacity,
+                        rule: fillRule)
```

Comment:
Now with the required parameters, we'll need to still supply values when nil...
```suggestion
            return Fill(
                color: fillColor ?? "black",
                opacity: fillOpacity ?? 1.0,
                rule: fillRule ?? .nonZero
            )
```

This would also be a good opportunity to perform the "none" check you referred to...

```swift
if let color = fillColor, color.caseInsensitiveCompare("none") == .orderedSame {
    return nil
}
```

Reply:
Having to supply the defaults at the point of calling is exactly what I tried to prevent.
As normally when any of the properties is nil you want the default value. With this change you need to remember the correct value when using the struct.